lucene/core/src/java/org/apache/lucene/util/automaton/RegExp.java

```diff
@@ -499,10 +508,29 @@ public RegExp(String s) throws IllegalArgumentException {
    *              regular expression
    */
   public RegExp(String s, int syntax_flags) throws IllegalArgumentException {
+    this(s, syntax_flags, 0);
+  }
+  /**
+   * Constructs new <code>RegExp</code> from a string.
+   * 
+   * @param s regexp string
+   * @param syntax_flags boolean 'or' of optional syntax constructs to be
+   *          enabled
+   * @param match_flags boolean 'or' of match behavior options such as case insensitivity
+   * @exception IllegalArgumentException if an error occurred while parsing the
+   *              regular expression
+   */
+  public RegExp(String s, int syntax_flags, int match_flags) throws IllegalArgumentException {    
+    // (for BWC reasons we don't validate invalid bits, just trim instead)
+    syntax_flags  = syntax_flags & 0xff;
```

Comment:
I don't think we need to maintain bw compat for this, is there any test that fails if you remove this line?

Reply:
As far as I can see, no. The change would be to remove that line and replace with

    if (syntax_flags >  ALL) {
      throw new IllegalArgumentException("Illegal syntax flag");
    }